### DISCUSSION TITLE
Option not appearing to protect who can push to a branch

### DISCUSSION THREAD

**Discussion by onwingslikeeagles** [community/community#23706] [New to GitHub]
<p>Hello,</p>
<p>I’ve read about protecting branches within GitHub and it seems the system has policies in place for this. What I want to achieve is for only certain individuals to post to particular branches. From what I’ve read <a href="https://help.github.com/en/articles/configuring-protected-branches" rel="nofollow noopener">here</a>, it seems this can be achieved through using the option <em>Restrict who can push to matching branches</em>.</p>
<p>I am not the owner of the repository, which means I can not view the settings for this repository, so I will be relaying this information to them. Therefore, in order to test this setting out, I looked at my own personal repository and I don’t see this option available. Both my repository and the one I’m working with are non-business accounts. Does this effect me not being able to see the option? In addition, I don’t have any collaborators against my personal repository to test this setting. Do I have to have at least one other person having access to the repository in order to view this setting?</p>
<p>Here is an attachment picture of what options are availble for adding a branch rule to my repository:  </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/992b185d977098471620f86ec458d6037421f227.png" data-download-href="/uploads/short-url/lQZnNZgtEv7npF1UU7EjA8X2ALl.png?dl=1" title="branch protection.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32335496/181078832-356a94d1-7154-4d37-b4da-dc43728d5442.png" alt="branch protection.PNG" data-base62-sha1="lQZnNZgtEv7npF1UU7EjA8X2ALl" width="552" height="499" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/992b185d977098471620f86ec458d6037421f227_2_552x499.png, https://aws1.discourse-cdn.com/github/optimized/1X/992b185d977098471620f86ec458d6037421f227_2_828x748.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/992b185d977098471620f86ec458d6037421f227.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/992b185d977098471620f86ec458d6037421f227_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">branch protection.PNG</span><span class="informations">938×849 39 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by lee-dohm** [comment, marked as answer]
<p>Yes, there are two separate features at work here: <a href="https://help.github.com/en/articles/configuring-protected-branches" rel="nofollow noopener">protected branches</a> and <a href="https://help.github.com/en/articles/enabling-branch-restrictions" rel="nofollow noopener">branch restrictions</a>. Branch restrictions is the feature that allows organization owners and people with admin permissions for organization-owned repositories to enforce branch restrictions so that only certain users or teams are able to push to a protected branch. See the links above for more information.</p>
<p>I hope that helps!</p>

**Comment by doxa-holdings** [comment]
<p>I have the same problem.</p>
<p>I have admin account of a Pro Github account but the options of “Restrict who can push to matching branches” doesn’t appear on my page as well.</p>
<p>Below is the screenshot.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2d91d2e8643908fbb84fdb5e9d77b13f859e4479.png" data-download-href="/uploads/short-url/6v7Uz3DEU7KLSt1letqVefzZGTn.png?dl=1" title="Screenshot 2019-11-06 13.26.15.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/50973714/181078835-758f0e4c-ea54-48d8-a524-ee620cd2f7b9.png" alt="Screenshot 2019-11-06 13.26.15.png" data-base62-sha1="6v7Uz3DEU7KLSt1letqVefzZGTn" width="690" height="469" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2d91d2e8643908fbb84fdb5e9d77b13f859e4479_2_690x469.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/2d91d2e8643908fbb84fdb5e9d77b13f859e4479_2_1035x703.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/2d91d2e8643908fbb84fdb5e9d77b13f859e4479_2_1380x938.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2d91d2e8643908fbb84fdb5e9d77b13f859e4479_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2019-11-06 13.26.15.png</span><span class="informations">2128×1448 319 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Could anyone help me with this matter?</p>
<p>Thanks in advance.</p>

**Comment by lee-dohm** [comment]
<p>Does the repository you’re attempting to modify belong to an organization? Or is it a repository in your personal account? As I stated above:</p>
<p>&gt; Branch restrictions is the feature that allows <strong>organization owners</strong> and people with admin permissions for <strong>organization-owned repositories</strong> to enforce branch restrictions so that only certain users or teams are able to push to a protected branch.</p>
<p>If you’re not working with a repository that belongs to an organization, branch restrictions will not be available.</p>
<p>I hope that helps!</p>

**Comment by rojasjandro89** [comment]
<p>Great, so I went to a Pro account just for nothing… thanks Github! I’ll make sure to downgrade my account ass soon as I hit the submit button :unamused:</p>
